pliers/filters/text.py

```diff
@@ -60,14 +63,40 @@ def __init__(self, stemmer='porter', tokenize=True, *args, **kwargs):
                              "instance of class StemmerI.")
         self.stemmer = stemmer
         self.tokenize = tokenize
+        self.case_sensitive = case_sensitive
         super(WordStemmingFilter, self).__init__()
 
     def _filter(self, stim):
+    
+        pos_map = {
+            'ADJ':'a',
+            'ADJ_SAT': 's',
+            'ADV': 'r',
+            'NOUN': 'n',
+            'VERB': 'v'
+        }
+    
+        def pos_wordnet(txt):
+            pos_tagged = dict(nltk.pos_tag(txt, tagset='universal'))
+            pos_tagged = {tok: pos_map[tag] if tag in pos_map else 'n' for tok,tag in pos_tagged.items()}
+            return pos_tagged
+        
         if self.tokenize:
```

Comment:
I could be missing something, but I think the conditional logic could be simplified. Can we handle the tokenization, lemmatization, and stemming steps serially, instead of branching? At least for the top-level conditional, I think you can just wrap `stim.text` in a list and then the same stemming/lemmatization logic can be applied regardless of tokenization.

Reply:
if I interpreted correctly, this should be fixed by latest commit.